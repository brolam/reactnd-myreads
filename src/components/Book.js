import React from 'react';
import PropTypes from 'prop-types'
import BookShelfChanger from './BookShelfChanger.js';
import BookShelfImage from './BookShelfImage.js';

class Book extends React.Component {
    render() {
        const { book, onChangeBookShelf } = this.props
        return (
            <li>
                <div className="book">
                    <div className="book-top">
                        <BookShelfImage bookId={book.id} />
                        <BookShelfChanger bookId={book.id} selectedOption={book.shelf} onChangeBookShelf={onChangeBookShelf} />
                    </div>
                    <div className="book-title">{book.title}</div>
                    <div className="book-authors">
                        {book.authors.map(function callback(author, index, authors) {
                            const separated = index > 0 ? '' : ','
                            return `${author}${separated} `
                        })}
                    </div>
                </div>
            </li>
        );
    }
}

Book.propTypes = {
    book: PropTypes.shape({
        id: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        authors: PropTypes.array.isRequired,
        shelf: PropTypes.string.isRequired,
    }),
    onChangeBookShelf: PropTypes.func.isRequired
}

export default Book