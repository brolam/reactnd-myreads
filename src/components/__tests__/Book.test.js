import React from 'react';
import Book from '../Book.js';
import {isBookObject} from '../Book.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

const linuxBook = {
    id:"nggnmAEACAAJ",
    title: "The Linux Command Line",
    authors: ["William E. Shotts", "Breno Marques" ],
    subtitle:"A Complete Introduction",
    shelf:"currentlyReading"
}
const onChangeBookShelf = () => {}

test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Book book={linuxBook} onChangeBookShelf={onChangeBookShelf} />, div)
})

test('Last Snapshot', () => {
    const book = renderer.create(
        <Book book={linuxBook} onChangeBookShelf={onChangeBookShelf} />
    );
    let tree = book.toJSON();
    expect(tree).toMatchSnapshot();
})

test('It is a book object', () => {
    let aBookObject = {
        id: "PGR2AwAAQBAJ",
        title: "To Kill a Mockingbird",
        authors: ["Harper Lee", "Breno Marques"],
        shelf: "currentlyReading"
      }
    expect(isBookObject(aBookObject)).toEqual(true);
})

test('It is not a book object', () => {
    expect(isBookObject({id: "*",title: "*"})).toEqual(false);
    expect(isBookObject({id: "*",authors: []})).toEqual(false);
    expect(isBookObject({title: "*",authors: []})).toEqual(false);
})