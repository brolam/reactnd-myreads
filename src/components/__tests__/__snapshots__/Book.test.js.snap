// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Snapshot 1`] = `
<li>
  <div
    className="book"
  >
    <div
      className="book-top"
    >
      <div
        className="book-cover"
        style={
          Object {
            "backgroundImage": "url(\\"https://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1\\")",
            "height": 190,
            "width": 120,
          }
        }
      />
      <div
        className="book-shelf-changer"
      >
        <select
          id="nggnmAEACAAJ"
          onChange={[Function]}
          value="currentlyReading"
        >
          <option
            disabled={true}
            value=""
          >
            Move to...
          </option>
          <option
            value="currentlyReading"
          >
            Currently Reading
          </option>
          <option
            value="wantToRead"
          >
            Want to Read
          </option>
          <option
            value="read"
          >
            Read
          </option>
          <option
            value="none"
          >
            None
          </option>
        </select>
      </div>
    </div>
    <div
      className="book-title"
    >
      The Linux Command Line
    </div>
    <div
      className="book-authors"
    >
      William E. Shotts, Breno Marques
    </div>
  </div>
</li>
`;
